sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala

```diff
@@ -43,12 +45,12 @@ import org.apache.spark.sql.types._
  *   2. Support DESC command.
  *   3. Support SHOW command.
  */
-@Ignore
 class ThriftServerQueryTestSuite extends SQLQueryTestSuite {
 
   private var hiveServer2: HiveThriftServer2 = _
 
-  override def beforeEach(): Unit = {
+  override def beforeAll(): Unit = {
+    super.beforeAll()
```

Comment:
In an earlier PR I commented that the flakiness may be because of async issues.
I meant that calling `startThriftServer` actually starts some things asynchronously in the background before the server becomes ready.
Moving it to `beforeAll` instead of `beforeEach` should help the flakiness by having this race only at the start of the suite and not before every test, but I think adding a 3 or 5 s sleep would make sure it never happens.

Reply:
I seem to be hitting it in the first test of the suite quite often:
```
[info] ThriftServerQueryTestSuite:
[info] - query_regex_column.sql *** FAILED *** (184 milliseconds)
[info]   java.sql.SQLException: Could not open client transport with JDBC Uri: jdbc:hive2://localhost:15421: java.net.ConnectException: Connection refused (Connection refused)
```

Reply:
`pgSQL/text.sql` should fail if `spark.sql.hive.thriftServer.async` is enabled: https://github.com/apache/spark/pull/25567#issuecomment-524507309

Reply:
Anyway, let's try to enable it: https://github.com/apache/spark/pull/26172